tests/function_component_attr/autoprops-fail.stderr

```diff
@@ -0,0 +1,72 @@
+error: function components can't accept a receiver
+ --> tests/function_component_attr/autoprops-fail.rs:7:23
+  |
+7 | fn CantAcceptReceiver(&self, b: bool) -> Html {
+  |                       ^^^^^
+
+error: could not find #[function_component] attribute in function declaration (#[autoprops] must be placed *before* #[function_component])
```

Comment:
I think we cannot enforce a function_component macro by reading the macro name as `function_component` can be renamed with `as` and can be implemented via other macros like `stylist::yew::styled_component`.

Reply:
Done in e8fe2f0

Reply:
Though imo we should use a separate proc macro instead of outputting the function component from an extended macro. This allows better composition:

```
#[styled_component]
#[autoprops]
#[function_component]
fn MyStyledComponent() -> Html {
    html! {<div class={css!("color: red;")}>{"Hello World!"}</div>}
}
```

Reply:
yeah `#[styled] #[autoprops] #[component]` / `#[autoprops] #[styled] #[component]` ([yewstack/yew/#3447](https://github.com/yewstack/yew/pull/3447) would be a breaking change btw so stylist would need to be updated anyway) should compose well, but its up to @futursolo to make a breaking change just for that